test/xlsx-test.js

```diff
@@ -0,0 +1,156 @@
+import {test} from "tap";
+import {ExcelWorkbook} from "../src/xlsx.js";
+
+function mockWorkbook(contents) {
+  return {
+    worksheets: Object.keys(contents).map((name) => ({name})),
+    getWorksheet(name) {
+      const _rows = contents[name];
+      return {
+        _rows: _rows.map((row) => ({
+          _cells: row.map((cell) => ({value: cell})),
+          hasValues: !!row.length,
+        })),
+        rowCount: _rows.length,
+        columnCount: Math.max(..._rows.map((r) => r.length)),
+      };
+    },
+  };
+}
+
+test("FileAttachment.xlsx reads sheet names", (t) => {
+  const workbook = new ExcelWorkbook(mockWorkbook({Sheet1: []}));
+  t.same(workbook.sheetNames(), ["Sheet1"]);
+  t.end();
+});
+
+test("FileAttachment.xlsx reads sheets", (t) => {
+  const workbook = new ExcelWorkbook(
+    mockWorkbook({
+      Sheet1: [
+        ["one", "two", "three"],
+        [1, 2, 3],
+      ],
+    })
+  );
+  t.same(workbook.sheet(0), [
+    {A: "one", B: "two", C: "three"},
+    {A: 1, B: 2, C: 3},
+  ]);
+  t.end();
+});
+
+test("FileAttachment.xlsx reads sheets with different types", (t) => {
+  const workbook = new ExcelWorkbook(
+    mockWorkbook({
+      Sheet1: [
+        ["one", {richText: [{text: "two"}, {text: "three"}]}],
+        [
+          {text: "link", hyperlink: "https://example.com"},
+          2,
+          {formula: "=B2*5", result: 10},
+        ],
+      ],
+    })
+  );
+  t.same(workbook.sheet(0), [
+    {A: "one", B: "twothree"},
+    {A: `<a href="https://example.com">link</a>`, B: 2, C: 10},
+  ]);
+  t.end();
+});
+
+test("FileAttachment.xlsx reads sheets with headers", (t) => {
+  const workbook = new ExcelWorkbook(
+    mockWorkbook({
+      Sheet1: [
+        ["one", "one", "one", "two"],
+        [1, null, 3, 4],
+        [5, 6, 7, 8],
+      ],
+    })
+  );
+  t.same(workbook.sheet(0, {headers: true}), [
+    {one: 1, one_: 3, two: 4},
+    {one: 5, one__: 6, one_: 7, two: 8},
```

Comment:
This one is particularly sneaky; the layout of keys gets inverted.

Reply:
agreed. I think maybe if there's a header row I can seed all of the column names using that row?